zerver/webhooks/semaphore2/view.py

```diff
@@ -0,0 +1,180 @@
+# Webhooks for external integrations.
+from typing import Any, Dict, List
+
+from django.http import HttpRequest, HttpResponse
+
+from zerver.decorator import api_key_only_webhook_view
+from zerver.lib.request import REQ, has_request_variables
+from zerver.lib.response import json_success
+from zerver.lib.webhooks.common import check_send_webhook_message
+from zerver.models import UserProfile
+from urllib.parse import urlparse
+
+# There seems to be a possibility of only 2 types of events : Push or a
+# Pull request, which can trigger a pipeline.
+
+# As per the Semaphore2.0 documentation, notifications would be sent only when
+# a pipeline is triggered.
+
+PUSH_TEMPLATE = """
+**{email}** pushed to branch **{branch_name}** of \
```

Comment:
Would it be better to use the login/username instead of the email? The payload doesn't seem to have name.

Reply:
I wasn't sure that login is username or the name of the user. We just have to use an unique field and if login is the username of the user then we can definitely use that.

Reply:
I suspect it would be the username. You could play around with the Semaphore API and verify.